index.js

```diff
@@ -42,22 +56,38 @@ function walkSync (baseDir, _options, _relativePath) {
   var entries = fs.readdirSync(baseDir + '/' + relativePath).sort();
   for (var i = 0; i < entries.length; i++) {
     var entryRelativePath = relativePath + entries[i];
-    var stats = getStat(baseDir + '/' + entryRelativePath);
+    var fullPath = baseDir + '/' + entryRelativePath;
+    var stats = getStat(fullPath);
 
     if (stats && stats.isDirectory()) {
       if (options.directories !== false && (!m || m.match(entryRelativePath))) {
-        results.push(entryRelativePath + '/');
+        results.push(new Entry(relativePath, stats.mode, stats.size, stats.mtime.getTime()));
```

Comment:
> This changed entryRelativePath + '/' to relativePath - test failure is related I think. Similar in L69 below.

good catch.



> Performance is much better indeed, good enough in my opinion.

Ya, I agree. Thanks for catching this regression, i didn't even notice the benchmark :P